src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackServiceTest.java

```diff
@@ -158,6 +164,158 @@ void should_not_allow_to_process_callback_in_case_transformation_url_not_configu
+            EVENT_ID,
+            caseDetails,
+            ignoreWarnings
+        ), IDAM_TOKEN, USER_ID);
+
+        // then
+        if (ignoreWarnings) {
+            assertThat(output.isRight()).isTrue();
+            assertThat(output.get().getModifiedFields().keySet()).containsOnly("caseReference");
+        } else {
+            assertThat(output.isLeft()).isTrue();
+            assertThat(output.getLeft()).containsOnly("some warning");
+        }
+    }
+
+    @Test
+    void should_report_error_if_classification_new_application_with_documents_and_without_ocr_data()
+        throws IOException, CaseTransformationException {
+        // given
+        setUpTransformationUrl();
+
+        Map<String, Object> data = new HashMap<>();
+        // putting 6 via `ImmutableMap` is available from Java 9
+        data.put("poBox", "12345");
+        data.put("journeyClassification", NEW_APPLICATION.name());
+        data.put("formType", "Form1");
+        data.put("deliveryDate", "2019-09-06T15:30:03.000Z");
+        data.put("openingDate", "2019-09-06T15:30:04.000Z");
+        data.put("scannedDocuments", TestCaseBuilder.document("https://url", "some doc"));
+
+        CaseDetails caseDetails = TestCaseBuilder.createCaseWith(builder -> builder
+            .id(1L)
+            .caseTypeId(CASE_TYPE_ID)
+            .jurisdiction("some jurisdiction")
+            .data(data)
+        );
+
+        // when
+        Either<List<String>, ProcessResult> output = service.process(new CcdCallbackRequest(
+            EVENT_ID,
+            caseDetails,
+            true
+        ), IDAM_TOKEN, USER_ID);
+
+        // then
+        assertThat(output.isLeft()).isTrue();
+        assertThat(output.getLeft()).containsOnly(
+            "Event createCase not allowed for the current journey classification NEW_APPLICATION without OCR"
+        );
+    }
+
+    @Test
+    void should_create_exception_record_classification_exception_with_documents_and_ocr_data_for_transformation_client()
+        throws IOException, CaseTransformationException {
+        // given
+        setUpTransformationUrl();
+        when(s2sTokenGenerator.generate()).thenReturn(randomUUID().toString());
+        when(transformationClient.transformExceptionRecord(eq("url"), any(ExceptionRecord.class), anyString()))
+            .thenReturn(new SuccessfulTransformationResponse(null, emptyList()));
+
+        Map<String, Object> data = new HashMap<>();
+        // putting 6 via `ImmutableMap` is available from Java 9
+        data.put("poBox", "12345");
+        data.put("journeyClassification", EXCEPTION.name());
+        data.put("formType", "Form1");
+        data.put("deliveryDate", "2019-09-06T15:30:03.000Z");
+        data.put("openingDate", "2019-09-06T15:30:04.000Z");
+        data.put("scannedDocuments", TestCaseBuilder.document("https://url", "some doc"));
+        data.put("scanOCRData", TestCaseBuilder.ocrDataEntry("some key", "some value"));
```

Comment:
building the `data` map is repeated 3 times. Perhaps it could be extracted to a method

Reply:
@rkondratowicz, this is part of my clean-up task left behind intentionally